Mathlib/CategoryTheory/Galois/GaloisObjects.lean

```diff
@@ -95,6 +95,106 @@ instance isPretransitive_of_isGalois (X : C) [IsGalois X] :
+    ∃! (τ : Aut B), f ≫ τ.hom = σ.hom ≫ f := by
+  let F := GaloisCategory.getFiberFunctor C
+  obtain ⟨a⟩ := nonempty_fiber_of_isConnected F A
+  refine ⟨?_, ?_, ?_⟩
+  · exact (evaluationEquivOfIsGalois F B (F.map f a)).symm (F.map (σ.hom ≫ f) a)
+  · apply evaluation_injective_of_isConnected F A B a
+    simp
+  · intro τ hτ
+    apply evaluation_aut_injective_of_isConnected F B (F.map f a)
+    simpa using congr_fun (F.congr_map hτ) a
+
+/-- A morphism from a connected object to a Galois object induces a map on automorphism
+groups. This is a group homomorphism (see `autMapMul`). -/
+noncomputable def autMap {A B : C} [IsConnected A] [IsGalois B] (f : A ⟶ B) (σ : Aut A) :
+    Aut B :=
+  (exists_autMap f σ).choose
+
+@[simp]
+lemma comp_autMap {A B : C} [IsConnected A] [IsGalois B] (f : A ⟶ B) (σ : Aut A) :
+    f ≫ (autMap f σ).hom = σ.hom ≫ f :=
+  (exists_autMap f σ).choose_spec.left
+
+@[simp]
+lemma comp_autMap_apply {A B : C} [IsConnected A] [IsGalois B] (f : A ⟶ B) (σ : Aut A)
+    (a : F.obj A) :
+    F.map (autMap f σ).hom (F.map f a) = F.map f (F.map σ.hom a) := by
+  simpa [-comp_autMap] using congrFun (F.congr_map (comp_autMap f σ)) a
+
+/-- `autMap` is uniquely characterized by making the canonical diagram commute. -/
+lemma autMap_unique {A B : C} [IsConnected A] [IsGalois B] (f : A ⟶ B) (σ : Aut A)
+    (τ : Aut B) (h : f ≫ τ.hom = σ.hom ≫ f) :
+    autMap f σ = τ :=
+  ((exists_autMap f σ).choose_spec.right τ h).symm
+
+@[simp]
+lemma autMap_id {A : C} [IsGalois A] : autMap (𝟙 A) = id :=
+  funext fun σ ↦ autMap_unique (𝟙 A) σ _ (by simp)
+
+@[simp]
+lemma autMap_comp {X Y Z : C} [IsConnected X] [IsGalois Y] [IsGalois Z] (f : X ⟶ Y)
+    (g : Y ⟶ Z) : autMap (f ≫ g) = autMap g ∘ autMap f := by
+  refine funext fun σ ↦ autMap_unique _ σ _ ?_
+  rw [Function.comp_apply, Category.assoc, comp_autMap, ← Category.assoc]
+  simp
+
+/-- `autMap` is surjective, if the source is also Galois. -/
+lemma autMap_surjective_of_isGalois {A B : C} [IsGalois A] [IsGalois B] (f : A ⟶ B) :
+    Function.Surjective (autMap f) := by
+  intro σ
+  let F := GaloisCategory.getFiberFunctor C
+  obtain ⟨a⟩ := nonempty_fiber_of_isConnected F A
+  obtain ⟨a', ha'⟩ := surjective_of_nonempty_fiber_of_isConnected F f (F.map σ.hom (F.map f a))
+  obtain ⟨τ, (hτ : F.map τ.hom a = a')⟩ := MulAction.exists_smul_eq (Aut A) a a'
+  use τ
+  apply evaluation_aut_injective_of_isConnected F B (F.map f a)
+  simp [hτ, ha']
+
+@[simp]
+lemma autMap_apply_mul {A B : C} [IsConnected A] [IsGalois B] (f : A ⟶ B) (σ τ : Aut A) :
+    autMap f (σ * τ) = autMap f σ * autMap f τ := by
+  let F := GaloisCategory.getFiberFunctor C
+  obtain ⟨a⟩ := nonempty_fiber_of_isConnected F A
+  apply evaluation_aut_injective_of_isConnected F (B : C) (F.map f a)
+  simp [Aut.Aut_mul_def]
+
+/-- `MonoidHom` version of `autMap`. -/
+noncomputable def autMapMul {A B : C} [IsConnected A] [IsGalois B] (f : A ⟶ B) :
```

Comment:
There should be an equational lemma for `autMapMul` (a better name might be `autMapHom`?) in order to ease the definition of `autGaloisSystem`. Adding `@[simps]` or `@[simps apply]`? may give the correct lemma.

Reply:
`@[simps!]` worked.